extensions/data-transfer/portability-data-transfer-imgur/src/main/java/org/datatransferproject/datatransfer/imgur/photos/ImgurPhotosExporter.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright 2019 The Data Transfer Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.datatransferproject.datatransfer.imgur.photos;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.datatransferproject.api.launcher.Monitor;
+import org.datatransferproject.spi.transfer.provider.ExportResult;
+import org.datatransferproject.spi.transfer.provider.Exporter;
+import org.datatransferproject.types.common.ExportInformation;
+
+import org.datatransferproject.types.transfer.auth.TokensAndUrlAuthData;
+import org.datatransferproject.types.common.models.photos.PhotoAlbum;
+import org.datatransferproject.types.common.models.photos.PhotoModel;
+import org.datatransferproject.types.common.models.photos.PhotosContainerResource;
+
+import java.io.IOException;
+import java.util.*;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+
+/** Exports Imgur albums and photos using Imgur API */
+public class ImgurPhotosExporter
+    implements Exporter<TokensAndUrlAuthData, PhotosContainerResource> {
+  private final String albumPhotosUrlTemplate;
+  private final String allPhotosUrlTemplate;
+  private final String albumsUrlTemplate;
+  private final String defaultAlbumId;
+
+  private final OkHttpClient client;
+  private final ObjectMapper objectMapper;
+  private Monitor monitor;
+
+  public ImgurPhotosExporter(Monitor monitor, String baseUrl) {
+    albumPhotosUrlTemplate = baseUrl + "/album/%s/images";
```

Comment:
Since we dont expect the basUrl of the API to change, can we make all these final static and initialize them as constants above?

Reply:
Done